src/main/java/ludogorie_soft/reservations_platform_api/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+
+import lombok.AllArgsConstructor;
+import ludogorie_soft.reservations_platform_api.dto.LoginDto;
+import ludogorie_soft.reservations_platform_api.dto.RegisterDto;
+import ludogorie_soft.reservations_platform_api.entity.Role;
+import ludogorie_soft.reservations_platform_api.entity.User;
+import ludogorie_soft.reservations_platform_api.exceotion.APIException;
+import ludogorie_soft.reservations_platform_api.mapper.UserMapper;
+import ludogorie_soft.reservations_platform_api.repository.RoleRepository;
+import ludogorie_soft.reservations_platform_api.repository.UserRepository;
+import ludogorie_soft.reservations_platform_api.service.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserMapper userMapper;
+    private UserRepository userRepository;
+    private RoleRepository roleRepository;
+    private PasswordEncoder passwordEncoder;
+
+    @Override
+    public String register(RegisterDto registerDto) {
+        if (!registerDto.getPassword().equals(registerDto.getRepeatPassword())) {
+            throw new APIException(HttpStatus.BAD_REQUEST, "Passwords do not match!");
+        }
+
+        if (userRepository.existsByUsername(registerDto.getName())) {
+            throw new APIException(HttpStatus.BAD_REQUEST, "Username already exists!");
+        }
+
+        if (userRepository.existsByEmail(registerDto.getEmail())) {
+            throw new APIException(HttpStatus.BAD_REQUEST, "Email already exists!");
+        }
+
+        User user = userMapper.toEntity(registerDto);
+
+        Set<Role> roles = new HashSet<>();
+
+        Optional<Role> optionalUserRole = Optional.ofNullable(roleRepository.findByName("ROLE_USER"));
+
+        Role userRole;
+        if (optionalUserRole.isPresent()) {
+            userRole = optionalUserRole.get();
+        } else {
+            userRole = new Role();
+            userRole.setName("ROLE_USER");
+            roleRepository.save(userRole);
+        }
+
+        roles.add(userRole);
+        user.setRoles(roles);
+
+        userRepository.save(user);
+
+        return "User Registered Successfully!";
+    }
+    @Override
+    public String login(LoginDto loginDto) {
+        String usernameOrEmail = loginDto.getUsernameOrEmail();
+        String password = loginDto.getPassword();
+        Optional<User> userOptional = userRepository.findByUsernameOrEmail(usernameOrEmail, usernameOrEmail);
+        if (!userOptional.isPresent()) {
+            return "Invalid username or email";
+        }
+
+        User user = userOptional.get();
+        if (!passwordEncoder.matches(password, user.getPassword())) {
+            return "Invalid password";
+        }
+        return "Login successful";
+    }
```

Comment:
I think it is better that the login returns as a response information about the logged-in user so that it can be recognized in FE